### PR TITLE
[X86][GlobalISel] Reorganize shift scalar tests (NFC)

### PR DESCRIPTION
**Pull request by e-kud**
Removed duplicated tests from GlobalISel directory

### CODE REVIEW

**Comment by e-kud** [comment]
Basically, moved `llvm/test/CodeGen/X86/fast-isel-shift.ll` to `llvm/test/CodeGen/X86/isel-shift.ll` with checking all three selectors and removed duplicates from `llvm/test/CodeGen/X86/GlobalISel`.

File: llvm/test/CodeGen/X86/isel-shift.ll [hunk 1]
```diff
@@ -0,0 +1,756 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc < %s -mtriple=x86_64-apple-darwin10 | FileCheck %s --check-prefixes=COMMON,SDAG
+; RUN: llc < %s -fast-isel -fast-isel-abort=1 -mtriple=x86_64-apple-darwin10 | FileCheck %s --check-prefixes=COMMON,FASTISEL
+; RUN: llc < %s -global-isel -global-isel-abort=1 -mtriple=x86_64-apple-darwin10 | FileCheck %s --check-prefixes=COMMON,GISEL
```
Comment (RKSimon) [inline llvm/test/CodeGen/X86/isel-shift.ll:4]: Would it be safe to add i686 coverage? I'm not sure if all gisel legalization is in place or not.

Comment (e-kud) [reply llvm/test/CodeGen/X86/isel-shift.ll:4]: Hmm, yeah, there is a problem with `G_ICMP` and `i64` input. Probably it'd look weird: reorganizing shuffle tests with `G_ICMP` legalization.
BTW, do you mean `i686` only for `GlobalISel` or for all of them?

Comment (RKSimon) [reply llvm/test/CodeGen/X86/isel-shift.ll:4]: Adding i686 for all of them - but if GlobalIsel is currently failing, then just the other two (with a TODO to add GlobalISel i686 tests in the future).

Comment (e-kud) [reply llvm/test/CodeGen/X86/isel-shift.ll]: Tunnel shifts are required to support `i64` shifts for `i686`. I already have  the implementation on Phabricator. So there is a todo for now.

**Review by RKSimon — APPROVED** [review]
LGTM

**Pull request merged**